In [6]:
import candidateSet

# UserFilter

In [7]:
NIR1 = candidateSet.uiMatrix("ml-100k_NIR1.json", USER=1, RATING=0)
# NIR0 = candidateSet.uiMatrix("ml-100k_NIR0.json", USER=1)

In [3]:
simi = ["cosine", "euclidean", "jaccard" , "manhattan"]

In [4]:
similaritys = candidateSet.Similarity(NIR1.uiMatrix, ["cosine", "jaccard" , "manhattan"])

/Users/DGW/miniforge3/lib/python3.9/site-packages/sklearn/metrics/pairwise.py:1975: DataConversionWarning: Data was converted to boolean for metric jaccard
  warnings.warn(msg, DataConversionWarning)


In [5]:
results_data_15 = []
num_u = 13
total_i = 20
count = 0
total = 0
for i in range(len(NIR1.uiMatrix)):
    elem = NIR1.uiData[i]
    
    candidate_items = NIR1.getCandidate(i, similaritys.getSimilaritys(i), 13, 20, 0)

    if elem[-1] in candidate_items:
        count += 1
    total +=1
print (f'count/total:{count}/{total}={count*1.0/total}')
print ('-----------------\n')

count/total:170/943=0.18027571580063625
-----------------



# ItemFilter

In [2]:
NIR1 = candidateSet.uiMatrix("ml-100k_NIR1.json", USER=0, RATING=1)
# NIR0 = candidateSet.uiMatrix("ml-100k_NIR0.json", USER=1)

In [4]:
simi = ["cosine", "euclidean", "jaccard" , "manhattan"]

In [31]:
similaritys = candidateSet.Similarity(NIR1.uiMatrix, ["cosine", "euclidean", "jaccard" , "manhattan"])

/Users/DGW/miniforge3/lib/python3.9/site-packages/sklearn/metrics/pairwise.py:1975: DataConversionWarning: Data was converted to boolean for metric jaccard
  warnings.warn(msg, DataConversionWarning)


In [32]:
results_data_15 = []
num_u = 13
total_i = 20
count = 0
total = 0
for i in range(len(NIR1.uiData)):
    elem = NIR1.uiData[i]
    
    candidate_items = NIR1.getCandidate(i, similaritys.getSimilaritysAll(), 13, 20, 0)

    if elem[-1] in candidate_items:
        count += 1
    total +=1
print (f'count/total:{count}/{total}={count*1.0/total}')
print ('-----------------\n')

count/total:105/943=0.11134676564156946
-----------------



## UserFilter
### RATING = 0

+ *baseline: count/total:170/943=0.180* & count/total:74/170=0.435(NIR_Method)


+ Cosine: count/total:172/943=0.182
+ Euclidean: count/total:170/943=0.180
+ **Jaccard: count/total:175/943=0.186** & **count/total:97/175=0.554(NIR_Method)** & count/total:89/175=0.509  & count/total:92/175=0.526(exchange) & count/total:89/175=0.509(exchange+modify)
+ Manhattan: count/total:170/943=0.180


+ Cosine+Euclidean: count/total:171/943=0.181
+ **Cosine+Jaccard: count/total:174/943=0.185** & 
+ Cosine+Manhattan: count/total:169/943=0.179
+ Euclidean+Jaccard: count/total:173/943=0.183
+ Euclidean+Manhattan: count/total:170/943=0.180
+ Jaccard+Manhattan: count/total:172/943=0.182


+ Cosine+Euclidean+Jaccard: count/total:170/943=0.180
+ Cosine+Euclidean+Manhattan: count/total:177/943=0.188
+ Cosine+Jaccard+Manhattan: count/total:170/943=0.180
+ **Euclidean+Jaccard+Manhattan: count/total:178/943=0.189** & count/total:89/175=0.508 & 


+ Cosine+Euclidean+Jaccard+Manhattan: count/total:176/943=0.187 & count/total:92/176=0.523(NIR_Method)

### RATING = 1
+ Cosine: count/total:163/943=0.172
+ Euclidean: count/total:147/943=0.156
+ **Jaccard: count/total:175/943=0.186** & count/total:93/175=0.531
+ Manhattan: count/total:146/943=0.155


+ Cosine+Euclidean: count/total:159/943=0.169
+ **Cosine+Jaccard: count/total:169/943=0.179** & count/total:87/169=0.514
+ Cosine+Manhattan: count/total:159/943=0.169
+ Euclidean+Jaccard: count/total:163/943=0.173
+ Euclidean+Manhattan: count/total:153/943=0.162
+ Jaccard+Manhattan: count/total:167/943=0.177


+ Cosine+Euclidean+Jaccard: count/total:169/943=0.180
+ Cosine+Euclidean+Manhattan: count/total:160/943=0.170
+ **Cosine+Jaccard+Manhattan: count/total:170/943=0.180** count/total:87/170=0.511
+ Euclidean+Jaccard+Manhattan: count/total:162/943=0.172


+ Cosine+Euclidean+Jaccard+Manhattan: count/total:165/943=0.175 & count/total:90/165=0.545


## ItemFilter
### RATING = 0

+ **Cosine: count/total:166/943=0.176**
+ Euclidean: count/total:24/943=0.025
+ Jaccard: count/total:167/943=0.177
+ Manhattan: count/total:24/943=0.025


+ Cosine+Euclidean: count/total:109/943=0.116
+ **Cosine+Jaccard: count/total:169/943=0.179**
+ Cosine+Manhattan: count/total:109/943=0.116
+ Euclidean+Jaccard: count/total:111/943=0.117
+ Euclidean+Manhattan: count/total:24/943=0.025
+ Jaccard+Manhattan: count/total:110/943=0.117


+ **Cosine+Euclidean+Jaccard: count/total:147/943=0.156**
+ Cosine+Euclidean+Manhattan: count/total:69/943=0.073
+ Cosine+Jaccard+Manhattan: count/total:146/943=0.155
+ Euclidean+Jaccard+Manhattan: count/total:67/943=0.071


+ Cosine+Euclidean+Jaccard+Manhattan: count/total:109/943=0.116

### RATING = 1
+ **Cosine: count/total:171/943=0.181**
+ Euclidean: count/total:32/943=0.034
+ Jaccard: count/total:167/943=0.177
+ Manhattan: count/total:10/943=0.011


+ Cosine+Euclidean: count/total:113/943=0.120
+ **Cosine+Jaccard: count/total:176/943=0.187**
+ Cosine+Manhattan: count/total:96/943=0.102
+ Euclidean+Jaccard: count/total:115/943=0.122
+ Euclidean+Manhattan: count/total:15/943=0.016
+ Jaccard+Manhattan: count/total:96/943=0.102


+ **Cosine+Euclidean+Jaccard: count/total:156/943=0.165**
+ Cosine+Euclidean+Manhattan: count/total:63/943=0.067
+ Cosine+Jaccard+Manhattan： count/total:138/943=0.146
+ Euclidean+Jaccard+Manhattan: count/total:67/943=0.071


+ Cosine+Euclidean+Jaccard+Manhattan: count/total:105/943=0.111



# Baseline Test

In [ ]:
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

In [133]:
!python3 main.py --RATING 0  --similarity "jaccard" --length_limit 15 --API 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 --saveFile "jaccard_fewShot_exPrompt.json"






/Users/DGW/miniforge3/lib/python3.9/site-packages/sklearn/metrics/pairwise.py:1975: DataConversionWarning: Data was converted to boolean for metric jaccard
  warnings.warn(msg, DataConversionWarning)
count/total:167/943=0.17709437963944857
-----------------

Using GPT
GT:Clear and Present Danger
Refine Summary:Based on the movies you have watched, it seems that you enjoy action, sci-fi, and adventure genres. You also seem to prefer movies with strong characters and engaging storylines. Additionally, you have shown a preference for movies with high production value and popular franchises such as Star Trek.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Die Hard: With a Vengeance - Die Hard 2
2. Jurassic Park - Independence Day
3. The Terminator - Terminator 2: Judgment Day
4. Full Metal Jacket - Platoon
5. Star Trek: First Contact - Star Trek: Generations
6. In the Line of Fire - Clear and Present Danger
7. The Last of th

GT:Raising Arizona
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including comedy, drama, and thriller. You also seem to enjoy classic movies as well as newer releases. It's possible that you prioritize well-written and engaging plots, memorable characters, and strong performances from the actors.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. The Silence of the Lambs - The Hunt for Red October
2. Psycho - Rear Window
3. The Sting - The Blues Brothers
4. Shine - The Piano
5. Butch Cassidy and the Sundance Kid - Raiders of the Lost Ark
6. Indiana Jones and the Last Crusade - Back to the Future
7. Annie Hall - Four Weddings and a Funeral
8. When Harry Met Sally... - Much Ado About Nothing
9. The Santa Clause - Muriel's Wedding
10. Sleepless in Seattle - Sabrina

Note: These recommendations are based on similarities in genre, themes, and/or style.
PID:64; count/total:3/11=0.272

GT:Crash
Refine Summary:Based on the movies you have watched, it seems that you enjoy a variety of genres and are open to trying new things. You may also appreciate movies with strong plotlines and interesting characters.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Boogie Nights - The Devil's Advocate
2. Chasing Amy - In & Out
3. The Big Lebowski - Wag the Dog
4. Starship Troopers - The Peacemaker
5. Contact - Gattaca
6. Scream - Kiss the Girls
7. Midnight in the Garden of Good and Evil - The Devil's Own
8. Murder at 1600 - Conspiracy Theory
9. Air Force One - Tomorrow Never Dies
10. Dante's Peak - Volcano
PID:106; count/total:9/20=0.45

GT:The Devil's Advocate
Refine Summary:Based on the movies you have watched, it seems that you prefer dramas and thrillers with complex characters and intricate plots. You also seem to enjoy movies with strong performances and critical acclaim. Overall, you seem to value quality story

Prompting Wrong:  That model is currently overloaded with other requests. You can retry your request, or contact us through our help center at help.openai.com if the error persists. (Please include the request ID cf2d3b6ede05750d547c1bcd5b3eed04 in your message.)
GT:Rosewood
Refine Summary:Based on the movies you have watched, it is difficult to determine your specific preferences. However, some possible features that may be important to you when selecting movies could be action, drama, suspense, adventure, comedy, and possibly some elements of romance.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Kiss the Girls - The Devil's Advocate
2. The Peacemaker - Starship Troopers
3. Seven Years in Tibet - The Wings of the Dove
4. In & Out - Leave It to Beaver
5. Soul Food - Eve's Bayou
6. Cop Land - Red Corner
7. George of the Jungle - The Saint
8. Event Horizon - Alien: Resurrection
9. Smile Like Yours - Titanic
10. A Thousan

GT:L.A. Confidential
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including drama, comedy, mystery, and historical films. You also seem to appreciate movies with strong characters and well-developed plots. It is possible that you prioritize quality over popularity when selecting movies.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. GoodFellas - L.A. Confidential
2. As Good As It Gets - The Full Monty
3. The Good, The Bad and The Ugly - Cape Fear
4. Titanic - The English Patient
5. Kolya - Gandhi
6. Manhattan - Sense and Sensibility
7. Manhattan Murder Mystery - The Game
8. Searching for Bobby Fischer - Chasing Amy
9. Ridicule - Seven Years in Tibet
10. Out to Sea - The Long Kiss Goodnight

Note: These recommendations are based on similarities in genre, themes, and/or style.
PID:206; count/total:20/38=0.5263157894736842

GT:The Silence of the Lambs
Refine Summary:Based on t

GT:Lone Star
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of drama, romance, and comedy genres. You also seem to appreciate movies with strong character development and thought-provoking themes. Additionally, you have watched some critically acclaimed and award-winning films, indicating that you value quality and artistic merit in your movie choices.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Fargo - The Big Night
2. Dead Man Walking - The People vs. Larry Flynt
3. Leaving Las Vegas - Jerry Maguire
4. Sense and Sensibility - Emma
5. Jane Eyre - Breaking the Waves
6. Ridicule - Everyone Says I Love You
7. Looking for Richard - Lone Star
8. Trainspotting - Swingers
9. Restoration - The Godfather
10. Il Postino - Secrets & Lies
PID:251; count/total:25/47=0.5319148936170213

GT:Murder at 1600
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres

GT:Volcano
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including thriller, action, and sci-fi. You may also prefer movies with a strong plot and interesting characters. It's possible that you enjoy movies with unexpected twists and turns, and that you appreciate good cinematography and special effects.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. The Game - L.A. Confidential
2. The Devil's Advocate - The Peacemaker
3. Starship Troopers - Volcano
4. Air Force One - The Peacemaker
5. Sphere - Event Horizon
6. Lost Highway - Midnight in the Garden of Good and Evil
7. Dante's Peak - Volcano
8. Event Horizon - Alien: Resurrection
9. Murder at 1600 - L.A. Confidential
10. Conspiracy Theory - Chasing Amy
PID:334; count/total:30/57=0.5263157894736842

GT:The Rock
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of action, drama, and thriller gen

GT:Jaws 2
Refine Summary:Based on the movies you have watched, it seems that you enjoy horror, thriller, and science fiction genres. You may also prefer movies with a supernatural or psychological element.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Nightmare on Elm Street - Hellraiser: Bloodline
2. The Amityville Horror - Amityville II: The Possession
3. In the Mouth of Madness - Bad Taste
4. The Omen - The Believers
5. The Prophecy - Wolf
6. Village of the Damned - Cat People
7. The Birds - The Fog
8. The Abyss - Body Parts
9. Jaws - Jaws 2
10. Natural Born Killers - Serial Mom
PID:366; count/total:37/67=0.5522388059701493

GT:Liar Liar
Refine Summary:Based on the movies you have watched, it seems that you enjoy a variety of genres and are open to trying new things. You may also appreciate strong performances from actors and engaging storylines.
predictions:Sure, here are 10 movies from the Candidate Set that are si

GT:Postino, Il
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of classic and contemporary films, with a preference for dramas and character-driven stories. You also seem to appreciate films with strong performances and well-crafted storytelling.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Citizen Kane - The Godfather
2. Fargo - A Fish Called Wanda
3. Blade Runner - The Empire Strikes Back
4. The Graduate - Chasing Amy
5. Annie Hall - Leaving Las Vegas
6. Sunset Blvd. - The Silence of the Lambs
7. The Philadelphia Story - His Girl Friday
8. Apollo 13 - Contact
9. Singin' in the Rain - Evita
10. Return of the Jedi - Star Trek: First Contact
PID:418; count/total:40/76=0.5263157894736842

GT:Psycho
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of classic and contemporary films from various genres. You may appreciate movies with strong storytelling, m

GT:Groundhog Day
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of classic and contemporary films, with a preference for dramas and comedies. You also seem to appreciate movies with strong characters and engaging storylines. It's possible that you value movies that are thought-provoking or emotionally impactful, but also enjoy lighter fare that provides a good laugh.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you've watched:

1. To Kill a Mockingbird - The Shawshank Redemption
2. The Great Dictator - The Blues Brothers
3. Annie Hall - This Is Spinal Tap
4. Back to the Future - Young Frankenstein
5. A Streetcar Named Desire - Cool Hand Luke
6. The Graduate - Heathers
7. Schindler's List - The Godfather
8. Dumb & Dumber - Monty Python and the Holy Grail
9. Sleepless in Seattle - Casablanca
10. Get Shorty - Pulp Fiction
PID:480; count/total:46/85=0.5411764705882353

Prompting Wrong:  That model is currentl

GT:When Harry Met Sally...
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of classic and contemporary films, with a preference for drama and comedy genres. You also seem to appreciate movies with strong character development and compelling storylines.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Casablanca - The Sting
2. Blade Runner - Twelve Monkeys
3. The Sound of Music - The Wizard of Oz
4. M*A*S*H - The Fugitive
5. Raising Arizona - A Fish Called Wanda
6. Jerry Maguire - Forrest Gump
7. Gandhi - The Shawshank Redemption
8. Back to the Future - Raiders of the Lost Ark
9. Room with a View - Monty Python and the Holy Grail
10. The Bridge on the River Kwai - Aliens
PID:537; count/total:50/94=0.5319148936170213

GT:The Full Monty
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of drama, biographical, and historical films. You may also appreciate movi

GT:Carrie
Refine Summary:Based on the movies you have watched, it seems that you enjoy horror and thriller genres. You may also prefer classic horror movies and those with a supernatural or paranormal theme. It is possible that you also enjoy movies with suspenseful plots and unexpected twists.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. The Shining - Bram Stoker's Dracula
2. Candyman - Mary Shelley's Frankenstein
3. The Omen - The Prophecy
4. The Howling - An American Werewolf in London
5. The Amityville Horror - The Fog
6. Wes Craven's New Nightmare - Nightmare on Elm Street
7. Scream - Village of the Damned
8. Evil Dead II - Lord of Illusions
9. The Birds - Cat People
10. Army of Darkness - Tales From the Crypt Presents: Demon Knight
PID:603; count/total:55/104=0.5288461538461539

GT:Dave
Refine Summary:Based on the movies you have watched, it seems that you enjoy comedies, romantic comedies, and classic films. You

GT:The Rock
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres, including drama, comedy, action, and romance. You may also appreciate movies with strong character development and engaging storylines. It's possible that you prioritize movies with well-known actors or directors, as some of the movies you have watched feature big names in the industry.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Fargo - The People vs. Larry Flynt
2. Leaving Las Vegas - Donnie Brasco
3. Heat - The Rock
4. Grosse Pointe Blank - Jerry Maguire
5. Men in Black - Austin Powers: International Man of Mystery
6. Sabrina - Mighty Aphrodite
7. That Thing You Do! - Tin Cup
8. Willy Wonka and the Chocolate Factory - Mystery Science Theater 3000: The Movie
9. Private Parts - The Truth About Cats & Dogs
10. Swingers - Chasing Amy

Note: These recommendations are based on similarities in genre, themes, and/or ac

GT:Good Will Hunting
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including comedy, action, thriller, and drama. You may also prefer popular or well-known movies, as many of the movies you have watched are box office hits.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Titanic - The English Patient
2. Air Force One - The Devil's Own
3. The Game - The Devil's Advocate
4. Conspiracy Theory - L.A. Confidential
5. Tomorrow Never Dies - Starship Troopers
6. Murder at 1600 - G.I. Jane
7. The Jackal - Dante's Peak
8. The Wedding Singer - Evita
9. Blues Brothers 2000 - Wag the Dog
10. Bean - In & Out
PID:728; count/total:66/123=0.5365853658536586

GT:Brazil
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of classic and contemporary films, with a preference for dramas and biopics. You also seem to appreciate movies with strong storytelling and char

GT:Jerry Maguire
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of action, drama, comedy, and thriller genres. You may also prefer movies with popular actors and actresses, as well as those with compelling storylines and high production value.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. The Godfather - The Rock
2. Men in Black - Mars Attacks!
3. Mission: Impossible - Executive Decision
4. Air Force One - Executive Decision
5. Sling Blade - Dead Man Walking
6. Evita - The Hunchback of Notre Dame
7. The Birdcage - My Best Friend's Wedding
8. Mr. Holland's Opus - A
9. Tomorrow Never Dies - Titanic
10. Eraser - Conspiracy Theory
PID:799; count/total:71/133=0.5338345864661654

GT:Tales from the Crypt Presents: Bordello of Blood
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including action, horror, thriller, and classic films. You may also a

GT:Alien: Resurrection
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including drama, action, thriller, and romance. You may also be drawn to popular or critically acclaimed movies. It is difficult to determine specific features that are most important to you without more information, but it appears that you are open to a variety of movie options.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Titanic - The Peacemaker
2. L.A. Confidential - The Rainmaker
3. As Good As It Gets - Good Will Hunting
4. Starship Troopers - G.I. Jane
5. Scream - Kiss the Girls
6. Tomorrow Never Dies - Air Force One
7. Evita - Everyone Says I Love You
8. Lost Highway - Seven Years in Tibet
9. Mimic - Alien: Resurrection
10. Conspiracy Theory - The Devil's Own
PID:819; count/total:76/142=0.5352112676056338

GT:Down Periscope
Refine Summary:Based on the movies you have watched, it seems that you enjo

GT:Raiders of the Lost Ark
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of classic and contemporary films, with a preference for dramas and romantic comedies. You may also appreciate movies with strong character development and engaging storylines.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. To Kill a Mockingbird - The Princess Bride
2. Little Women - Muriel's Wedding
3. Mr. Holland's Opus - Dances with Wolves
4. Bringing Up Baby - Sabrina
5. The Man Who Would Be King - Raiders of the Lost Ark
6. Jaws - Groundhog Day
7. Somewhere in Time - Return of the Jedi
8. Manon of the Spring - Dances with Wolves
9. How to Make an American Quilt - Winnie the Pooh and the Blustery Day
10. Koyaanisqatsi - Pulp Fiction
PID:877; count/total:82/151=0.543046357615894

GT:The Truth About Cats & Dogs
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres includin

GT:Chasing Amy
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including drama, thriller, action, and suspense. You may also appreciate movies with strong performances from actors and well-written plots. It's possible that you enjoy movies that challenge you intellectually or emotionally, and that you are open to exploring different themes and subject matter.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. The Game - The Jackal
2. Gattaca - Event Horizon
3. Contact - As Good As It Gets
4. The Devil's Advocate - Deconstructing Harry
5. Boogie Nights - The Full Monty
6. Air Force One - Murder at 1600
7. Conspiracy Theory - Apt Pupil
8. The Devil's Own - Cop Land
9. Wag the Dog - Starship Troopers
10. Hoodlum - Alien: Resurrection

Note: These recommendations are based on similarities in genre, themes, and/or style, and may not necessarily be exact matches.
PID:919; count/total:86

In [137]:
!python3 main.py --RATING 0 --filename "ml_100k.json"  --similarity "jaccard" --API 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 --saveFile "jaccard_fewShot_non.json"






/Users/DGW/miniforge3/lib/python3.9/site-packages/sklearn/metrics/pairwise.py:1975: DataConversionWarning: Data was converted to boolean for metric jaccard
  warnings.warn(msg, DataConversionWarning)
count/total:169/943=0.1792152704135737
-----------------

Using GPT
GT:Under Siege 2: Dark Territory
Refine Summary:Based on the movies you have watched, it seems that you enjoy action movies with a lot of suspense and thrill. You may also prefer movies with a strong lead character and a fast-paced plot.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. True Lies - GoldenEye
2. Speed - Terminal Velocity
3. Last Man Standing - Young Guns
4. Demolition Man - Waterworld
5. Eraser - Clear and Present Danger
6. Last Action Hero - The Shadow
7. Cliffhanger - In the Line of Fire
8. Star Trek: The Motion Picture - Star Trek IV: The Voyage Home
9. Die Hard 2 - Under Siege 2: Dark Territory
10. Batman Forever - The Crow
PID:16; count/tot

GT:Tales from the Crypt Presents: Bordello of Blood
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including horror, thriller, sci-fi, and classic films. You may also appreciate movies with complex plots and unexpected twists.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Twelve Monkeys - In the Mouth of Madness
2. The Game - The Believers
3. Mimic - The Howling
4. Body Snatchers - The Invasion of the Body Snatchers
5. Screamers - Hellraiser: Bloodline
6. Once Upon a Time in the West - The Good, the Bad and the Ugly
7. The Sting - Fargo
8. Children of the Corn: The Gathering - April Fool's Day
9. The Howling - The Fog
10. In the Mouth of Madness - Lord of Illusions
PID:96; count/total:5/10=0.5

GT:Conspiracy Theory
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of action, drama, and suspense genres. It is difficult to determine your specif

GT:Fantasia
Refine Summary:Based on the movies you have watched, it seems that you enjoy a variety of genres and styles. You may appreciate movies with strong characters and engaging storylines, as well as those with elements of romance, action, and comedy. You may also be drawn to movies with cultural or historical significance. Overall, it appears that you value diversity and quality in your movie selections.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Forrest Gump - The Sound of Music
2. The Princess Bride - The Magnificent Seven
3. The Usual Suspects - The Fugitive
4. When Harry Met Sally... - True Lies
5. Return of the Jedi - Star Trek: First Contact
6. Rumble in the Bronx - Get Shorty
7. Much Ado About Nothing - Shakespeare in Love
8. Farewell My Concubine - The Hunt for Red October
9. The Silence of the Lambs - Heat
10. Terminator 2: Judgment Day - Alien
PID:156; count/total:12/20=0.6

GT:Unforgiven
Refine Summ

GT:The Devil's Own
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including action, comedy, horror, and drama. You may also have a preference for classic or cult films. It is difficult to determine specific features that are most important to you without more information, but it appears that you are open to a variety of styles and themes.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Die Hard: With a Vengeance - The Rock
2. Good Will Hunting - Jerry Maguire
3. Monty Python and the Holy Grail - The Game
4. The Manchurian Candidate - Murder at 1600
5. Con Air - Face/Off
6. Young Frankenstein - Wag the Dog
7. The Amityville Horror - The Devil's Own
8. Star Trek IV: The Voyage Home - Independence Day (ID4)
9. Twister - Dante's Peak
10. Liar Liar - The Saint

Note: These recommendations are based on similarities in genre, themes, and/or style.
PID:199; count/total:19/30=0.6333333

GT:Jurassic Park
Refine Summary:Based on the movies you have watched, it seems that you enjoy classic films, dramas, and adventure movies. You may also appreciate movies with strong character development and a compelling storyline.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Gone with the Wind - Schindler's List
2. Rear Window - North by Northwest
3. Glory - The Fugitive
4. Groundhog Day - Four Weddings and a Funeral
5. Dances with Wolves - Jurassic Park
6. Jumanji - The Blues Brothers
7. Ben-Hur - Star Trek: The Motion Picture
8. The Man Without a Face - The Silence of the Lambs
9. When Harry Met Sally... - Sleepless in Seattle
10. Jurassic Park - Apollo 13
PID:240; count/total:24/39=0.6153846153846154

GT:Postino, Il
Refine Summary:Based on the movies you have watched, it seems that you enjoy classic films, romantic comedies, and dramas. You may also appreciate movies with strong character development and compelling

GT:Scream
Refine Summary:Based on the movies you have watched, it is difficult to determine your specific preferences. However, it seems that you enjoy a mix of action, comedy, and crime/thriller genres. It is possible that you prioritize entertainment value and a good plot when selecting movies.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Con Air - Independence Day (ID4)
2. Ransom - Primal Fear
3. Austin Powers: International Man of Mystery - The Fifth Element
4. Don't Be a Menace to South Central While Drinking Your Juice in the Hood - The People vs. Larry Flynt
5. Jackie Chan's First Strike - Mission: Impossible
6. Night Falls on Manhattan - Liar Liar
7. Beavis and Butt-head Do America - Mighty Aphrodite
8. Kids in the Hall: Brain Candy - Dead Man Walking
9. Scream - Bound
10. Tin Cup - Jerry Maguire

Note: These recommendations are based on genre and themes, and may not necessarily have the same exact plot or char

GT:Raising Arizona
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including drama, romance, and comedy. You may also appreciate movies with strong character development and emotional depth. It is possible that you prioritize critically acclaimed films and those with compelling storylines.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Schindler's List - Gandhi
2. Dances with Wolves - On Golden Pond
3. Ghost - Casablanca
4. When Harry Met Sally... - Four Weddings and a Funeral
5. Back to the Future - Indiana Jones and the Last Crusade
6. Sleepless in Seattle - The Princess Bride
7. Kolya - One Flew Over the Cuckoo's Nest
8. Get Shorty - Pulp Fiction
9. Mary Poppins - E.T. the Extra-Terrestrial
10. Stand by Me - Butch Cassidy and the Sundance Kid
PID:354; count/total:32/58=0.5517241379310345

GT:Sleepless in Seattle
Refine Summary:Based on the movies you have watched, it seems 

GT:Fargo
Refine Summary:Based on the movies you have watched, it seems that you enjoy independent films and dramas. You may also have a preference for character-driven stories with complex relationships and emotional depth. It's possible that you appreciate films that explore themes of identity, sexuality, and personal growth.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Losing Chase - Dead Man Walking
2. Chasing Amy - Jerry Maguire
3. Sense and Sensibility - Emma
4. Welcome to the Dollhouse - The Full Monty
5. Blood & Wine - Donnie Brasco
6. Boot - Kolya
7. Das - The Crucible
8. Shanghai Triad - Cold Comfort Farm
9. The Full Monty - Big Night
10. Emma - Postino, Il
PID:406; count/total:37/67=0.5522388059701493

GT:Liar Liar
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including comedy, drama, and adventure. You may also appreciate movies with strong characters and engagi

GT:The Silence of the Lambs
Refine Summary:Based on the movies you have watched, it seems that you enjoy science fiction, action, drama, and classic films. You may also appreciate movies with strong characters and compelling storylines.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Aliens - Apollo 13
2. The Terminator - Die Hard
3. Star Trek: The Wrath of Khan - 2001: A Space Odyssey
4. The Right Stuff - Apollo 13
5. A Clockwork Orange - Dr. Strangelove or: How I Learned to Stop Worrying and Love the Bomb
6. The Godfather: Part II - GoodFellas
7. Casablanca - The Maltese Falcon
8. Mr. Smith Goes to Washington - The Graduate
9. The Shawshank Redemption - One Flew Over the Cuckoo's Nest
10. Citizen Kane - The Graduate
PID:471; count/total:43/77=0.5584415584415584

GT:Batman Returns
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres, including romantic comedies, action films, and c

GT:The First Wives Club
Refine Summary:Based on the movies you have watched, it is difficult to determine your specific preferences. However, some possible features that may be important to you when selecting movies could be a mix of drama and comedy, popular actors/actresses, and a variety of genres.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. My Best Friend's Wedding - The First Wives Club
2. The Truth About Cats & Dogs - Sense and Sensibility
3. That Old Feeling - Grumpier Old Men
4. The Mirror Has Two Faces - Mr. Holland's Opus
5. Broken Arrow - The Rock
6. Trial and Error - Conspiracy Theory
7. Nixon - The Godfather
8. The Lost World: Jurassic Park - Jurassic Park
9. My Best Friend's Wedding - Tin Cup
10. The Truth About Cats & Dogs - The Hunchback of Notre Dame
PID:523; count/total:52/87=0.5977011494252874

GT:Carrie
Refine Summary:Based on the movies you have watched, it seems that you enjoy horror and thriller

GT:The Lost World: Jurassic Park
Refine Summary:Based on the movies you have watched, it seems that you enjoy action, adventure, and thriller genres. You also seem to enjoy movies with popular actors and actresses. Therefore, it can be inferred that the most important features for you when selecting movies are the genre, the cast, and the overall entertainment value.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Independence Day (ID4) - The Lost World: Jurassic Park
2. The Hunt for Red October - Air Force One
3. Dante's Peak - Volcano
4. Indiana Jones and the Last Crusade - The Saint
5. Pretty Woman - Jerry Maguire
6. The Glimmer Man - Face/Off
7. Butch Cassidy and the Sundance Kid - Heat
8. Dragonheart - The Saint
9. The River Wild - The Perfect Storm
10. Liar Liar - Bruce Almighty

I hope you find these recommendations helpful!
PID:603; count/total:57/97=0.5876288659793815

GT:Evita
Refine Summary:Based on the movies 

GT:The Birdcage
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including action, adventure, comedy, and sci-fi/fantasy. You may also appreciate movies with well-known actors and a strong plot.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Star Trek: First Contact - Return of the Jedi
2. Mission: Impossible - Air Force One
3. Grumpier Old Men - Vegas Vacation
4. Executive Decision - Broken Arrow
5. Dragonheart - The Frighteners
6. Heaven's Prisoners - Absolute Power
7. The Ghost and the Darkness - Kingpin
8. Mystery Science Theater 3000: The Movie - Fargo
9. The Nutty Professor - Happy Gilmore
10. Tin Cup - Leaving Las Vegas
PID:645; count/total:62/106=0.5849056603773585

GT:Butch Cassidy and the Sundance Kid
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including action, drama, and comedy. You may also appreciate movies with str

PID:684; count/total:67/115=0.5826086956521739

GT:The Truth About Cats & Dogs
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including action, drama, and comedy. You may also be drawn to popular or well-known movies, as many of the movies you have watched are considered classics. Overall, it appears that you value a good storyline and engaging characters in your movie selections.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Star Wars - The Fifth Element
2. Independence Day (ID4) - Mission: Impossible
3. The Rock - Ransom
4. Toy Story - The Lost World: Jurassic Park
5. Mr. Holland's Opus - The First Wives Club
6. Leaving Las Vegas - Dead Man Walking
7. Executive Decision - Courage Under Fire
8. The Birdcage - The Truth About Cats & Dogs
9. Fargo - Twelve Monkeys
10. Tin Cup - Kingpin

Note: These recommendations are based on similarities in genre, themes, and/or overall ton

GT:Wag the Dog
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of drama, action, and thriller genres. It is difficult to determine your specific preferences based on this limited information, but it appears that you may be drawn to movies with strong characters and compelling storylines. You may also enjoy movies with a sense of adventure or suspense.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. The Truman Show - The Game
2. The Devil's Advocate - Conspiracy Theory
3. The Jackal - Tomorrow Never Dies
4. The Replacement Killers - The Saint
5. The Postman - Wag the Dog
6. The Apostle - Ulee's Gold
7. Welcome to Sarajevo - Kundun
8. Seven Years in Tibet - The Edge
9. The Game - The Sweet Hereafter
10. Conspiracy Theory - Chasing Amy

Note: These recommendations are based on genre, themes, and general similarities to the movies you have watched.
PID:739; count/total:76/126=0.6031746031746

GT:Indiana Jones and the Last Crusade
Refine Summary:Based on the movies you have watched, it seems that you enjoy classic films, as well as some action and adventure movies. You may also appreciate movies with strong characters and engaging storylines.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Casablanca - The Shawshank Redemption
2. Butch Cassidy and the Sundance Kid - Indiana Jones and the Last Crusade
3. The Sting - Ocean's Eleven
4. The Fugitive - The Hunt for Red October
5. Back to the Future - Raiders of the Lost Ark
6. Arsenic and Old Lace - Harold and Maude
7. The Quiet Man - North by Northwest
8. Sense and Sensibility - The Graduate
9. The Graduate - Annie Hall
10. North by Northwest - The Hunt for Red October
PID:792; count/total:82/136=0.6029411764705882

GT:Taxi Driver
Refine Summary:Based on the movies you have watched, it is difficult to determine your specific preferences. However, some possible feat

GT:Evita
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including action, thriller, and comedy. You may also prefer movies with well-known actors and actresses. It is difficult to determine your exact preferences based on a small sample size, but it appears that you enjoy movies with engaging plots and high production value.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Tomorrow Never Dies - The Devil's Own
2. The Jackal - The Devil's Advocate
3. Sphere - Gattaca
4. Conspiracy Theory - The Rainmaker
5. As Good As It Gets - The Full Monty
6. Starship Troopers - Volcano
7. The Saint - The Devil's Own
8. That Darn Cat! - Murder at 1600
9. That Darn Cat! - Liar Liar
10. Sphere - Lost Highway
PID:839; count/total:88/146=0.6027397260273972

GT:Cop Land
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres, including drama, comedy, and thrille

GT:The Saint
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including action, thriller, comedy, and drama. You may also be drawn to movies with well-known actors and actresses. It is difficult to determine specific features that are most important to you based on this limited information.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. The Jackal - The Saint
2. Conspiracy Theory - The Devil's Own
3. As Good As It Gets - Picture Perfect
4. Murder at 1600 - The Peacemaker
5. Midnight in the Garden of Good and Evil - L.A. Confidential
6. G.I. Jane - Evita
7. Flubber - The Full Monty
8. Scream - Kiss the Girls
9. The Devil's Own - Amistad
10. L.A. Confidential - Gattaca

Note: These recommendations are based on genre, themes, and actors/actresses.
PID:874; count/total:92/155=0.5935483870967742

GT:Waterworld
Refine Summary:Based on the movies you have watched, it seems that you en

GT:Liar Liar
Refine Summary:Based on the movies you have watched, it seems that you enjoy a mix of genres including action, thriller, and sci-fi. You may also prefer movies with well-known actors or actresses. It is difficult to determine specific features that are most important to you based on this limited information.
predictions:Sure, here are 10 movies from the Candidate Set that are similar to the movies you have watched:

1. Event Horizon - The Game
2. Hard Rain - Money Talks
3. Mimic - The Devil's Advocate
4. Fire Down Below - Cop Land
5. The Man Who Knew Too Little - The Saint
6. Deceiver - Apt Pupil
7. Jungle2Jungle - Fly Away Home
8. McHale's Navy - Hoodlum
9. Event Horizon - Spawn
10. Hard Rain - Wag the Dog

Note: These recommendations are based on genre and themes, and may not necessarily have the same actors or plotlines as the movies you have watched.
PID:911; count/total:99/165=0.6

GT:Liar Liar
Refine Summary:Based on the movies you have watched, it seems that you enj